.github/workflows/docker_firedrake-um2n.yml

```diff
@@ -0,0 +1,31 @@
+name: 'Build bespoke UM2N Docker container'
+
+on:
+  # Check that Docker build succeeds when these files get changed in an open PR
+  pull_request:
+    paths:
+      - '.github/workflows/docker_firedrake-um2n.yml'
+      - 'docker/Dockerfile.firedrake-um2n'
+
+  # Build and push the Docker container when these files get changed on the main branch
+  push:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/docker_firedrake-um2n.yml'
+      - 'docker/Dockerfile.firedrake-um2n'
+
+  # Manually trigger the workflow
+  workflow_dispatch:
```

Comment:
Sorry, I missed one more thing... I guess we want a `schedule` trigger too

Reply:
Hm I just noticed this is now missing from the Firedrake env, was that accidental? If so, I'll add it back in along with this one.

Reply:
It's on purpose. That's why I added `workflow_run`, so that the Firedrake workflow is run after the PETSc workflow completes. So Firedrake is effectively scheduled since PETSc is scheduled :)

Reply:
Ah okay got it. Added a schedule for UM2N Docker build in bde3d71.

Reply:
Thanks! But I think we should change the time here or in https://github.com/mesh-adaptation/UM2N/pull/59, so that the tests are run after the image is updated.